src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -81,6 +82,10 @@ public class FileInputStream extends InputStream
 
     private volatile boolean closed;
 
+    // This field indicates whether the position0() or skip0() may be
+    // invoked without encountering an illegal seek exception.
```

Comment:
You might be update this to say that it indicates if the file is regular file as some operations need the current position and to seek. Otherwise the src changes looks good.

Reply:
Updated in 98f918a.